src/bfd-disasm.cpp

```diff
@@ -1,132 +0,0 @@
-#include <fcntl.h>
-#include <memory>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <unistd.h>
-// bfd.h assumes everyone is using autotools and will error out unless
-// PACKAGE is defined. Some distros patch this check out.
-#define PACKAGE "bpftrace"
-#include <bfd.h>
-#include <dis-asm.h>
-#include <filesystem>
-
-#include "bfd-disasm.h"
-#include "util/system.h"
-
-namespace bpftrace {
-
-BfdDisasm::BfdDisasm(std::string &path)
-{
-  fd_ = open(path.c_str(), O_RDONLY);
-
-  if (fd_ >= 0) {
-    std::error_code ec;
-    std::filesystem::path fs_path{ path };
-    std::uintmax_t file_size = std::filesystem::file_size(fs_path, ec);
-
-    if (!ec) {
-      size_ = file_size;
-    }
-  }
-}
-
-BfdDisasm::~BfdDisasm()
-{
-  if (fd_ >= 0)
-    close(fd_);
-}
-
-static int fprintf_nop(void *out __attribute__((unused)),
-                       const char *fmt __attribute__((unused)),
-                       ...)
-{
-  return 0;
-}
-
-#ifdef LIBBFD_INIT_DISASM_INFO_FOUR_ARGS_SIGNATURE
-static int fprintf_styled_nop(void *out __attribute__((unused)),
-                              enum disassembler_style s __attribute__((unused)),
-                              const char *fmt __attribute__((unused)),
-                              ...)
-{
-  return 0;
-}
-#endif
-
-static AlignState is_aligned_buf(void *buf, uint64_t size, uint64_t offset)
-{
-  disassembler_ftype disassemble;
-  struct disassemble_info info;
-  auto tpath = util::get_pid_exe("self");
-  if (!tpath) {
-    return AlignState::Fail;
-  }
-
-  bfd *bfdf;
-
-  bfdf = bfd_openr(tpath->c_str(), nullptr);
```

Comment:
@jordalgo @viktormalik I've posted this speculatively, but I still feel like I'm taking crazy pills. Since the moment this was checked in, it has used `/proc/self/exe` for the `bfd_openr` path, and `fd_` is not used anywhere in the actual disassembly function. It seems random whether the offset would have lined up or not. Did I miss something obvious? If not, I would propose that we drop this feature since no one noticed that it was broken it's entire existence 🤦 . However, I'm more than happy to be wrong and missing something dumb!

Reply:
This is before my time. I have no clue what this is and what it was even attempting to do lol.